internal/moduletest/graph/transform_state.go

```diff
@@ -94,9 +94,11 @@ func (t *TestStateTransformer) Transform(g *terraform.Graph) error {
 				if err != nil {
 					return fmt.Errorf("error retrieving state for state key %q from backend: error retrieving state manager: %w", key, err)
 				}
+				stmgr.RefreshState()
```

Comment:
Realising that I never read in the state from the state file (lol) made me realise that my tests weren't very effective previously. I've updated some tests in ways that detect that state is used or not (see non deterministic ID of test_resource being deterministic due to state in `TestTest_UseOfBackends_priorStateUsedByBackend`)